### ISSUE TITLE
Modifying training parameters causes the training results to become zero

### ISSUE THREAD

**Issue by ILZFLNO02** [roboflow/rf-detr#527]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

My dataset trains normally with the following training configuration:
```python
from rfdetr import RFDETRNano, RFDETRMedium, RFDETRSmall

if __name__ == "__main__":
    model = RFDETRSmall()

    model.train(
        dataset_dir="/oeecs/rf-detr/dataset/cj_12_month12_split_groups_y_break",
        epochs=150,
        # lr_drop=80,
        batch_size=6,
        num_workers=8,
        grad_accum_steps=1,
        lr=1e-4,
        output_dir="/oeecs/rf-detr/output/cj_12_month12_split_groups_y_break",
        amp=True,
        device="cuda",
        distributed=True,
        run_test=False,
        # num_queries=150,
        # num_select=150,
        # group_detr=3,
    )

``` 
<img width="1800" height="1200" alt="Image" src="https://github.com/user-attachments/assets/c3f3b24a-58df-49a3-8e88-0d4c083ef86e" />
Then I tried to modify some parameters to speed up training, but this resulted in very abnormal training behavior.
The modified training configuration is as follows:
```python
from rfdetr import RFDETRNano, RFDETRMedium, RFDETRSmall

if __name__ == "__main__":
    model = RFDETRSmall()

    model.train(
        dataset_dir="/oeecs/rf-detr/dataset/cj_12_month12_split_groups_y_break",
        epochs=150,
        # lr_drop=80,
        batch_size=6,
        num_workers=8,
        grad_accum_steps=1,
        lr=1e-4,
        output_dir="/oeecs/rf-detr/output/cj_12_month12_split_groups_y_break",
        amp=True,
        device="cuda",
        distributed=True,
        run_test=False,
        num_queries=150,
        num_select=150,
        group_detr=3,
    )

``` 
<img width="1800" height="1200" alt="Image" src="https://github.com/user-attachments/assets/1abe0c1d-2a14-4ebb-8ba0-bb5bb8ea50c0" />

The changes I made were setting:
num_queries = 150
num_select  = 150
group_detr  = 3

### Environment

python：3.9
gpu：2 of rtx3080

### Minimal Reproducible Example

These are the main parameters I modified.
You may try using them on your own dataset as a reference.
num_queries = 100
num_select  = 100
group_detr  = 3

### Additional

I have tried applying similar optimizations on other datasets.
With the following parameter settings:

num_queries = 100
num_select  = 100
group_detr  = 3

I was able to obtain very good results as well.
I am not sure why the results are sometimes very poor and sometimes very good.

### Are you willing to submit a PR?

- [x] Yes, I'd like to help by submitting a PR!

**Comment by ILZFLNO02** [comment]
The modifications to these parameters were mainly based on the reply in #416

**Comment by isaacrob-roboflow** [comment]
There may be a bug related to simultaneously reducing both group_detr and num_queries. Regardless, reducing num_queries from 200 to 150 won't get you much. Reducing group_detr to 3 from 13 will give big speedup and reduced memory but will usually lead to worse accuracy and may give unstable training.

**Comment by Borda** [comment]
Hi @ILZFLNO02 — apologies for not getting back to you sooner on this. v1.6 overhauled the training pipeline with PyTorch Lightning and fixed a `num_select` namespace regression where `TrainConfig` values silently overrode model-specific defaults. The interaction between `group_detr` and `num_queries` may behave differently now. We'd recommend upgrading to v1.6.1 and retesting. As @isaacrob noted, reducing num_queries from the default generally isn't beneficial.
